Shopify/ruby-lsp
# Add GIFs to our request documentation

### Motivation

Closes #108

Add GIFs to our request documentation to better show which features are supported.

### Implementation

By commit
1. Include the `misc` folder in YARD generation (assets just get copied over)
2. Exclude `misc` from the gemspec, so that users don't have to download the GIFs
3. Adapt `check_docs` to verify that, instead of written examples, the documentation includes a GIF
4. Change all request documentation to include GIFs
5. Add the `misc` folder with all of the request GIFs
6. Add a list of supported features in `RubyLsp::Requests` that we can point to from the README
7. Enforce that all requests are present in the `RubyLsp::Requests` list in `check_docs`

### Manual Tests

1. Switch to this branch
2. `bundle exec rake yard`
3. In a browser, use CMD + SHIFT + O and open the index file under `....../ruby-lsp/docs/index.html`
4. Navigate in the documentation for specific requests and check the GIFs

Alternatively, you can also start yard in server mode, but don't forget to pass it all of the options we have listed in the `Rakefile`.
